### ISSUE TITLE
Problem after updating to 1.1.1

### ISSUE THREAD

**Issue by Faugus** [Open-Wine-Components/ULWGL#213]
Users are reporting that after updating to 1.1.1, this problem happens:
```
WINEPREFIX=/home/wyktor/.config/faugus-launcher/prefixes GAMEID=wow PROTONPATH=GE-Proton9-15 "/usr/bin/umu-run" "/mnt/Games/battlenet/drive_c/Program Files (x86)/World of Warcraft/World of Warcraft Launcher.exe"
WARNING: '/home/wyktor/.local/share/umu/umu_version.json' is obsolete
WARNING: '/home/wyktor/.local/share/umu/reaper' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_dl_util.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_plugins.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_util.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_log.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_consts.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/umu_run.py' is obsolete
WARNING: '/home/wyktor/.local/share/umu/sniper_platform_0.20240125.75305' is obsolete
steamrt is up to date
pressure-vessel-wrap[73104]: W: /home/wyktor/.local/share/umu/var/tmp-LHAPV2/bin does not exist, this probably won't work
pressure-vessel-wrap[73104]: W: /home/wyktor/.local/share/umu/var/tmp-LHAPV2/etc does not exist, this probably won't work
pressure-vessel-wrap[73104]: W: /home/wyktor/.local/share/umu/var/tmp-LHAPV2/lib does not exist, this probably won't work
pressure-vessel-wrap[73104]: W: /home/wyktor/.local/share/umu/var/tmp-LHAPV2/sbin does not exist, this probably won't work
pressure-vessel-wrap[73104]: E: None of the supported CPU architectures are common to the graphics provider and the container (tried: x86_64-linux-gnu, i386-linux-gnu)
```
As a workaround, deleting the folder `/home/wyktor/.local/share/umu/` and running it again, fixed the issue.

**Comment by R1kaB3rN** [comment]
This problem is already fixed in main. See https://github.com/Open-Wine-Components/umu-launcher/commit/976ccfae0685fc4bd781e19d0cbf935d8016da17

**Comment by R1kaB3rN** [comment]
Actually, sorry. This is relevant to the variable directory, so that fix may not be relevant.

**Comment by R1kaB3rN** [comment]
The obsolete warnings can be ignored as that's just a signal to the user that they're useless. In the future, for major releases, that will change. Also, not to deny the existence of that error as I'm aware of posts similar to it, but I cannot reproduce those `pressure-vessel-wrap` errors when upgrading the runtime from `sniper_platform_0.20240125.75305` -> `sniper_platform_0.20241008.104210` when using the latest branch.

Here were my reproduction steps:

- Build the zipapp at https://github.com/Open-Wine-Components/umu-launcher/commit/459ba8efa7c3fbafd40f196bd925f6daadd3be57
- `rm -rf $HOME/.local/share/umu`
- `curl -LJO https://repo.steampowered.com/steamrt3/images/0.20240125.75305/SteamLinuxRuntime_sniper.tar.xz`
- `tar xaf SteamLinuxRuntime_sniper.tar.xz`
- `mkdir -p $HOME/.local/share/umu`
- `cp -rv SteamLinuxRuntime_sniper/* $HOME/.local/share/umu`
- `mv $HOME/.local/share/umu/_v2-entry-point $HOME/.local/share/umu/umu`
- `GAMEID=0 umu-run wineboot -u`

Though, if you're able to reproduce that error somehow, please list its steps as I'd be interested. Especially if it can be reproduced in main.

However, when upgrading from 0.1-RC4 to 1.1.1, what I can reproduce are some errors related to missing files in `$HOME/.local/share/umu/pressure-vessel` and this happened due to a regression in umu-launcher 1.1.1 where it was mistakenly performing both file move and remove operations in parallel on install. Again, this was addressed in https://github.com/Open-Wine-Components/umu-launcher/pull/207 and we have a test in our CI to mitigate this moving forward.

**Comment by R1kaB3rN** [comment]
For a more decisive solution, what we can do is just remove our entire runtime directory each update. This is the lazy but surefire way to avoid any variable directory related errors, since it's not exactly clear to me why those variable directory errors occur. Of course, it'll come at the risk of removing any user files in there, if we had users who were mistakenly creating files there.

**Comment by R1kaB3rN** [comment]
Closing as the removal of `$HOME/.local/share/umu/var` is fixed in https://github.com/Open-Wine-Components/umu-launcher/commit/cfe955f42b0fed574fd1f5bc4165f7491a775c25. Now, a clean copy of `var` will be created by pressure-vessel for each runtime update, so there should not be any errors related to file expectations in that directory.

For now, please inform your users to try to use the latest umu-launcher. Even with your suggested workaround, you're going to run into another error when the runtime updates when on version 1.1.1. 

Stay tuned for a hotfix that addresses these issues soon.

**Comment by Faugus** [comment]
Thanks @R1kaB3rN

**Comment by Faugus** [comment]
@R1kaB3rN I rebuilt the umu package on Fedora and now I'm getting this when I try to run something:
```
WINEPREFIX=/home/faugus/.config/faugus-launcher/prefixes/rufus GAMEID=rufus PROTONPATH=GE-Proton "/usr/bin/umu-run" "/home/faugus/Downloads/rufus-4.5p.exe"
GE-Proton is up to date
Updating steamrt to latest...
Downloading latest steamrt sniper, please wait...
SteamLinuxRuntime_sniper.tar.xz: SHA256 is OK
Traceback (most recent call last):
  File "/usr/lib64/python3.12/shutil.py", line 847, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: '/home/faugus/.cache/umu/tmpoh6lnsc5/SteamLinuxRuntime_sniper/run-in-sniper' -> '/home/faugus/.local/share/umu/run-in-sniper'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/umu-run", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.12/site-packages/umu/__main__.py", line 9, in main
    return umu_run()
           ^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/umu/umu_run.py", line 829, in main
    future.result()
  File "/usr/lib64/python3.12/concurrent/futures/_base.py", line 456, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/lib64/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/umu/umu_runtime.py", line 219, in setup_umu
    _update_umu(local, json, thread_pool, client_session)
  File "/usr/lib/python3.12/site-packages/umu/umu_runtime.py", line 385, in _update_umu
    _install_umu(json, thread_pool, client_session)
  File "/usr/lib/python3.12/site-packages/umu/umu_runtime.py", line 177, in _install_umu
    future.result()
  File "/usr/lib64/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/lib64/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/umu/umu_runtime.py", line 453, in _move
    move(src_file, dest_file)
  File "/usr/lib64/python3.12/shutil.py", line 867, in move
    copy_function(src, real_dst)
  File "/usr/lib64/python3.12/shutil.py", line 475, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib64/python3.12/shutil.py", line 260, in copyfile
    with open(src, 'rb') as fsrc:
```
When I try to run a second time:
```
WINEPREFIX=/home/faugus/.config/faugus-launcher/prefixes/rufus GAMEID=rufus PROTONPATH=GE-Proton "/usr/bin/umu-run" "/home/faugus/Downloads/rufus-4.5p.exe"
GE-Proton is up to date
steamrt is up to date
pressure-vessel-wrap[5264]: W: /home/faugus/.local/share/umu/var/tmp-XWA9U2/bin does not exist, this probably won't work
pressure-vessel-wrap[5264]: W: /home/faugus/.local/share/umu/var/tmp-XWA9U2/etc does not exist, this probably won't work
pressure-vessel-wrap[5264]: W: /home/faugus/.local/share/umu/var/tmp-XWA9U2/lib does not exist, this probably won't work
pressure-vessel-wrap[5264]: W: /home/faugus/.local/share/umu/var/tmp-XWA9U2/sbin does not exist, this probably won't work
pressure-vessel-wrap[5264]: E: None of the supported CPU architectures are common to the graphics provider and the container (tried: x86_64-linux-gnu, i386-linux-gnu)
```

**Comment by R1kaB3rN** [comment]
As I stated, that problem was already fixed in https://github.com/Open-Wine-Components/umu-launcher/commit/976ccfae0685fc4bd781e19d0cbf935d8016da17 and all stable versions will be effected until a hot fix is released. 

The reason the FileNotFoundError occurs when setting up or updating the runtime is because of a regression where umu-launcher was removing and installing files simultaneously. 

Please try reproduce this issue again with the zipapp that’s built at the latest main branch, and, if it still occurs, report back by creating another issue with the trace back after setting UMU_LOG=debug and the sha256 of the zipapp
